lib/neo4j/active_node/unpersisted.rb

```diff
@@ -7,22 +7,32 @@ def deferred_create_cache
         @deferred_create_cache ||= {}
       end
 
+      def deferred_options_cache
+        @deferred_options_cache ||= {}
+      end
+
       def defer_create(association_name, object, options = {})
         clear_deferred_nodes_for_association(association_name) if options[:clear]
 
+        deferred_options_for_association(association_name) << {verify: options[:verify]}
         deferred_nodes_for_association(association_name) << object
```

Comment:
I spent a few hours trying to get the `deferred_create_cache` to hold both the deferred nodes and options for each node (currently, options is just a hash which indicates whether the node's persistence should be verified, but in the future it could hold other stuff). My general (preferred) strategy would be `deferred_options_for_association(association_name) << [object, {verify: options[:verify]}]` rather than storing the options in a separate cache. I couldn't get it working though. Somewhere in the code base, the `@deferred_create_cache` is used for constructing associations in a manner I couldn't find. The following would fail:
```
person = Person.new
email = EmailAddress.new
person.email_addresses << email

person.email_addresses # <- this would get messed up by the `[object, {verify: options[:verify]}]` in the deferred_create_cache
```